Add Art.py and connectivity improvement

TheFab21 opened the pull request:
# Add comprehensive Frame Art Mode documentation and examples

## Overview
This PR adds complete documentation and configuration examples for Frame 
Art Mode functionality available in Samsung Frame TVs.

## What's Added

### Documentation (docs/)
- **Frame_Art.md** (19 KB) - Complete service documentation with 9 services, 
  parameters, examples, troubleshooting
- **Frame_Art_Gallery.md** (16 KB) - Interactive Lovelace gallery guide with 
  5 different layouts and customization options

### Examples (examples/frame_art/)
- **configuration.yaml** - Template sensors for galleries, input helpers
- **scripts.yaml** - 15 utility scripts (downloads, routines, artwork selection)
- **automations.yaml** - 20+ automation examples (time-based, motion, weather)
- **lovelace.yaml** - Complete interactive gallery dashboard
- **README.md** - Guide for using examples

### README Updates
- Added Frame Art in "Additional Features" list
- New comprehensive "Frame Art Mode" section with:
  - Quick start guide
  - Service examples
  - Compatible models
  - Troubleshooting
  - Links to detailed docs

## Features Documented
✅ Display artwork on Frame TV  
✅ Batch thumbnail downloads with smart caching  
✅ Interactive Lovelace galleries (click-to-display)  
✅ Brightness control (0-100)  
✅ Frame style management (mattes)  
✅ Slideshow control  
✅ Personal photos & Art Store support  
✅ 20+ automation examples  
✅ 15+ ready-to-use scripts  

## Testing
- Tested on Frame TV QE55LS03DAUXXN (2024 model)
- Tested on Home Assistant 2024.11
- All YAML examples validated
- All services tested and working
- Gallery layouts tested on desktop & mobile

## Benefits
- Complete documentation for Frame TV users
- Ready-to-copy examples save hours of configuration
- Interactive galleries enhance user experience
- Covers advanced use cases (automation, customization)

## Breaking Changes
None - purely additive (new docs & examples)

## Checklist
- [x] Documentation complete and accurate
- [x] YAML examples tested and working
- [x] No breaking changes
- [x] Links between docs verified
- [x] Code follows project style
- [x] Ready for review

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Merge pull request #1 from TheFab21/TheFab21-patch-1

Smartthings hangs on the TV

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Merge pull request #2 from TheFab21/TheFab21-patch-2

Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Merge pull request #3 from TheFab21/TheFab21-patch-3

The fab21 patch 3

TheFab21 committed:
Add files via upload

TheFab21 committed:
Remove set_art_mode service from Samsung TV integration

Removed 'set_art_mode' service from services.yaml. duplicated  service with art_set_artmode

TheFab21 committed:
Delete FRAME_ART_API_DOCUMENTATION.md

not accurate see complete guide in docs

TheFab21 committed:
Revise frame options and descriptions in Frame_Art.md

Updated frame settings and descriptions in documentation.

TheFab21 committed:
Insert image in Frame Art Gallery documentation

Added an image to the Frame Art Gallery documentation.

TheFab21 committed:
Clean of duplicated service

set_art_mode:
  name: Set Art Mode
  description: Send to samsung TV the command to set art mode.
  fields:
    entity_id:
      name: Entity Name
      description: Name of the target entity
      required: true
      example: "media_player.tv"
      selector:
        entity:
          integration: samsungtv_smart

TheFab21 committed:
Add async state to the switch art mode

TheFab21 committed:
Switch on TV, if an art mode action is performed aand TV is off

TheFab21 committed:
Add anti-flicker logic for folder sensor updates

Implement anti-flicker logic to prevent unnecessary re-renders when the file_list attribute remains unchanged.

chansearrington commented:
@ollo69 can we get this merged please!

TheFab21 commented:
Hi after a lot of testing,I've done a lot of corrections.. I'll complete my PR : this custom component will be able to work independently (oauth is managed). Just wait for a couple of weeks.

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Delete docs/Frame_Art.md

TheFab21 committed:
Delete docs/Frame_Art_Gallery.md

TheFab21 commented:
> @ollo69 can we get this merged please!

you can go on my repo and give a try ;-)

chansearrington commented:
@TheFab21 how do I try this out and make hacs point to your repo and not this one?

chansearrington committed:
Fix HACS installation - disable zip_release

Since no GitHub Releases are published, HACS fails to download the
integration with a 404 error when zip_release is true.

This change allows HACS to download directly from the repository
source instead of requiring a release ZIP file.

TheFab21 committed:
Merge pull request #4 from chansearrington/fix-hacs-installation

Fix HACS installation - disable zip_release

chadBonadonna commented:
@TheFab21  can you confirm if this still works with the Smart Things Developer Workspace being deprecated 
I went the new developer center site and was not able to follow the instructions in your readme to get it set up in home assistant :( 
<img width="1251" height="205" alt="Screenshot 2026-03-23 at 9 00 38 PM" src="https://github.com/user-attachments/assets/27174673-3bfb-4daa-ab06-da2cbc02aede" />

TheFab21 committed:
Add files via upload

TheFab21 commented:
I think the job is done , give a try to my repo and let me know, the (my) samsung integration is mainly to manage the frame TV and art mode  : [README_OAUTH2_EN.md](https://github.com/user-attachments/files/26208160/README_OAUTH2_EN.md)

chadBonadonna commented:
@TheFab21 , I really appreciate the reply and am excited to help test this but when I log into https://developer.smartthings.com/ it brings me to SmartThings Developer Center It does not have any options for a new project or "Automation for The Home" seems I only have option for creating a new Product which seems very different that what should need to be done. Right?

<img width="1412" height="676" alt="Screenshot 2026-03-24 at 6 14 38 PM" src="https://github.com/user-attachments/assets/7e6fb6d8-a8c8-4b79-8f9f-9670b1307346" />

TheFab21 commented:
Just install the custom component from my repo and configure it. No needs to have a dev Account, when you add the custom component in Ha you just add your TV ip and you will be redirected to the Samsung login page (the same way the official component works)

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Delete custom_components/samsungtv_smart/__pycache__ directory

TheFab21 committed:
Delete custom_components/samsungtv_smart/README.md

TheFab21 committed:
Delete custom_components/samsungtv_smart/README_CORRECTIONS.md

TheFab21 committed:
Delete custom_components/samsungtv_smart/README_OAUTH2.md

TheFab21 committed:
Delete custom_components/samsungtv_smart/README_OAUTH2_EN.md

TheFab21 committed:
Add files via upload

TheFab21 committed:
Improve media_player state handling in switch.py

Enhance availability and power state checks for TV.

TheFab21 committed:
Refactor event filter for media player state changes

TheFab21 committed:
Refactor power switch to use SmartThings API

Refactor Samsung TV power switch to use SmartThings API for turning off the TV directly. Added support for SmartThings configuration and improved logging.

TheFab21 committed:
Merge pull request #7 from TheFab21/Merge-entities

Merge entities

TheFab21 committed:
Revise OAuth2 setup instructions in README

Updated OAuth2 authentication instructions and added SmartThings CLI steps.

TheFab21 committed:
Remove OAuth2 token refresh troubleshooting steps

Removed troubleshooting steps for 'Token refresh failed' under OAuth2 section.

TheFab21 committed:
Revise OAuth2 section and clean up troubleshooting list

Updated section titles and removed unnecessary troubleshooting steps.

TheFab21 committed:
Revise OAuth2 setup instructions in README

Updated OAuth2 setup instructions and clarified steps for creating a SmartThings OAuth app. Adjusted formatting for consistency and improved readability.

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add Samsung Frame TV Art Mode switch entity

TheFab21 committed:
Bump version from 6.3.2 to 6.3.3

TheFab21 committed:
Update version badge to 6.3.3

TheFab21 committed:
Remove duplicate docstring in switch.py

Removed duplicate docstring for Samsung Frame TV Art Mode switch entity.

TheFab21 committed:
Update switch.py

TheFab21 committed:
Bump version from 6.3.3 to 6.3.4

TheFab21 committed:
Update version badge to 6.3.4

TheFab21 committed:
Refactor Samsung TV switch setup and clean imports

Removed unused imports and refactored the switch setup for Samsung TV integration.

TheFab21 committed:
Bump version from 6.3.4 to 6.3.5

TheFab21 committed:
Update version badge to 6.3.5

chadBonadonna commented:
@TheFab21  you are a legend, make sure to add one of those BuyMeACoffee links! https://buymeacoffee.com

thank you for the help! I think I am finally on my way of controlling this horrible TV 😂

TheFab21 commented:
Thanks a lot, read the other read the others readme for the custom card (pretty cool look)![image](https://github.com/user-attachments/assets/aa032294-c518-4652-b1a8-cc802d073fb9)
You can see your favorites for example, and select directly your frame image

TheFab21 committed:
Add files via upload

TheFab21 committed:
Bump version from 6.3.5 to 6.3.6

TheFab21 committed:
Update version badge to 6.3.6

TheFab21 committed:
Add files via upload

TheFab21 committed:
Add files via upload

TheFab21 committed:
Update version badge to 6.4

TheFab21 commented:
> @TheFab21 you are a legend, make sure to add one of those BuyMeACoffee links! https://buymeacoffee.com
> 
> thank you for the help! I think I am finally on my way of controlling this horrible TV 😂

check out 6.4 is online ( frame matte type and coulor are now automaticly created, no needs to add input select in conf anymore, and response time is improved)

chadBonadonna commented:
unfortunately I am not able to get the Art mode things to work, seemingly just power on and power off. but before even that was not reliable. Seemingly art mode is always unknown.


<img width="1560" height="1040" alt="Screenshot 2026-03-27 at 7 55 11 PM" src="https://github.com/user-attachments/assets/ae271879-8a85-45bb-a593-fbc39ae02ae8" />

TheFab21 commented:
Give a try to advanced option websocket ![image](https://github.com/user-attachments/assets/7b4a4537-7776-4e58-9884-c5f9b120cd4d)

TheFab21 committed:
Add guard for async_write_ha_state() call

Added guard to prevent async_write_ha_state() from being called before the entity is fully registered.

TheFab21 committed:
Bump version to 6.4.1 in manifest.json

TheFab21 committed:
Update version badge to 6.4.1

TheFab21 commented:
@chadBonadonna i've done a new realease 6.4.1, could you check

chadBonadonna commented:
Still having issues after release 6.7.1 - I can still turn the TV on and off reliably just not able to use any of the Frame/Art features. I dont expect you to continue to help troubleshoot as on/off is half the battle. but happy to troubleshoot it as much as you would want. 
<img width="1486" height="948" alt="Screenshot 2026-03-31 at 8 02 38 PM" src="https://github.com/user-attachments/assets/22185f2a-21bf-4cd5-a315-ab46b2cb4b1a" />
<img width="1262" height="1056" alt="Screenshot 2026-03-31 at 8 05 02 PM" src="https://github.com/user-attachments/assets/9af4c443-85c5-4a37-af90-e5003c127acb" />

TheFab21 commented:
Hi Chad 
In your configuration do 
logger:
  logs:
    custom_components.samsungtv_smart: debug
    custom_components.samsungtv_smart.api.art: debug

Restart HA


And send me the log, is your HA server on the same vlan ? If not use ip masquerade. It's a known limitation of websocket

TheFab21 commented:
And report issue on my repo, it will be easier as I have inforked from ollo69
